### PR TITLE
Fix printing of large values

### PR DESCRIPTION
**Pull request by saraedum**
the fixed size buffer of 20 bytes was not big enough for %f printing.
Instead we now rely on Arb's printing to produce reasonable outputs.

### CODE REVIEW

File: renf_elem/get_str_pretty.c [hunk 1]
```diff
@@ -41,11 +42,40 @@ char * renf_elem_get_str_pretty(renf_elem_t a, const char * var, renf_t nf, slon
...
+                // is too large for big numbers, %e is hard to read for small integers,
+                // %g does not show enough digits in many cases,) we use Arb's double
+                // printing instead.
+                {
+                    arb_t x;
+                    arb_init(x);
+
+                    arb_set_d(x, d);
+
+                    s = arb_get_str(x, 8, ARB_STR_NO_RADIUS);
```
Comment (saraedum) [inline renf_elem/get_str_pretty.c]: I chose to print 8 digits now, see below for an example why I think that this is a good choice.

File: renf_elem/test/t-get_str_pretty.c [hunk 2]
```diff
@@ -29,10 +29,20 @@ int main(void)
         EANTIC_STR_ALG | EANTIC_STR_ARB};
     char * output[5] = {
         "-2/3*x+1/2",
-        "-0.990712",
+        "-0.99071198",
```
Comment (saraedum) [inline renf_elem/test/t-get_str_pretty.c]: With 8 digits, the output here changes. Arguably, this does not add too much useful information here.

File: renf_elem/test/t-get_str_pretty.c [hunk 3]
```diff
@@ -29,10 +29,20 @@ int main(void)
...
-        "-2/3*x+1/2 ~ -0.990712",
+        "-2/3*x+1/2 ~ -0.99071198",
         "-2/3*x+1/2 ~ [-0.9907119850 +/- 1.43e-13]"};
+    char * shifted_output [8] = {
+        "-9.9071198",
+        "-99.071198",
+        "-990.71198",
+        "-9907.1198",
+        "-99071.198",
+        "-990711.98",
```
Comment (saraedum) [inline renf_elem/test/t-get_str_pretty.c]: With only 6 digits, I would get -9.90712e+5 here. I found that quite hard to parse already. However, I find that 7 digit integers are still easy to parse. So something like 9907119 still makes sense to me.

**Comment by videlec** [comment]
Your 8 digits default looks reasonable to me.

**Comment by videlec** [comment]
Some failures
```
FAIL: renfxx/test/t-get_str

===========================

terminate called after throwing an instance of 'std::runtime_error'

  what():  error with ALG | D, expected (0 ~ 0.000000) got (0 ~ 0)

FAIL renfxx/test/t-get_str (exit status: 134)

FAIL: renfxx/test/t-stream

==========================

terminate called after throwing an instance of 'std::runtime_error'

  what():  wrong string, got (A ~ 1.2599210)

FAIL renfxx/test/t-stream (exit status: 134)
```
